### DISCUSSION TITLE
How can I restrict some user roles to make changes to particular fields of a model.

### DISCUSSION THREAD

**Discussion by piyushdongre-dot-com** [filamentphp/filament#13939] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.57

### How can we help you?

Hi
Example I have a Clinic management system. I dont want user with role `staff` to change the `amount` field in record. Only `Doctor` or `Admin` can change that. I know by using policies I can restrict to make change on whole model, but how to do to only certain fields ?

Thanks

**Comment by leandrocfe** [comment, marked as answer]
`TextInput::make('amount')->disabled(!auth()->user()->is_admin)`
?

**Reply by piyushdongre-dot-com** [reply]
Yeah that works
